inst/include/common/fims_math.hpp

```diff
@@ -220,6 +268,420 @@ inline const Type ad_max(const Type &a, const Type &b, Type C = 1e-5) {
+     *  \f$\ \mu \f$  is the mean.<br>
+     *  \f$\ \theta\f$ is the standard deviation.<br>
+     * 
+     *  @image html http://upload.wikimedia.org/wikipedia/commons/7/74/Normal_Distribution_PDF.svg
+     * 
+     *  <br> <b>Source:</b> http://en.wikipedia.org/wiki/Normal_distribution<br>
+     * 
+     * 
+     * 
+     * @author Matthew Supernaw
+     * @date June 6, 2013
+     *  
+     * @param x
+     * @param mean
+     * @param sd
+     * @return relative likelihood for this random variable to have value x.
+     */
+    template<class T>
+    const T dnorm(const T &x, const T &mean, const T &sd, bool ret_log = false) {
+      
+        // Check if the standard deviation is valid
+        if (sd <= static_cast<T> (0)) {
+            throw std::invalid_argument("Standard deviation must be positive.");
+        }
+
+        // Constants
+        const T inv_sqrt_2pi = 1.0 / std::sqrt(2.0 * M_PI);
+
+        // Compute the exponent part
+        T z = (x - mean) / sd;
+        T exponent = -1.0*static_cast<T> (0.5) * z * z;
+
+        // Compute log density
+        if (ret_log) {
+            return exponent - log(sd) - log(inv_sqrt_2pi);
+        }
+
+        // Otherwise, return the density
+        return inv_sqrt_2pi / sd * exp(exponent);
+    }
+
+    /**
+     * @brief Normal distribution random sampling.
+     *
+     * @param mean The mean of the distribution \( \mu \).
+     * @param sd The standard deviation of the distribution \( \sigma \).
+     * @param gen Random number generator.
+     * @return T Random sample from the Normal distribution.
+     */
+    template<typename T>
+    T rnorm(T mean, T sd) {
+        static std::random_device rd;
+        static std::mt19937 gen(rd());
+        static std::uniform_real_distribution<double> dist(0.0, 1.0);
+
+        T u1 = static_cast<T> (dist(gen));
+        T u2 = static_cast<T> (dist(gen));
+
+        T z0 = fims_math::sqrt(-2.0 * log(u1)) * fims_math::cos(2.0 * M_PI * u2);
+
+        return z0 * sd + mean;
+    }
+
+    /**
+     * @ingroup LogNormal
+     * 
+     * @brief The probability distribution function for a log-normal distribution.
+     * 
+     * \f$
+     * f(x) = \frac{1}{x\sqrt{2\pi}\theta}e^{-\frac{(ln x - \mu)^{2}}{2\theta^{2}}}
+     * \f$
+     * 
+     * <br><b>Where,</b><br>
+     * \f$\mu\f$ is the log of the mean.<br>
+     * \f$\theta\f$ is the log of the standard deviation.<br>
+     * 
+     * @image html http://upload.wikimedia.org/wikipedia/commons/thumb/8/80/Some_log-normal_distributions.svg/500px-Some_log-normal_distributions.svg.png
+     * 
+     * <b><br>Source:</b>http://en.wikipedia.org/wiki/Log-normal_distribution<br><br>
+     * 
+     * 
+     * @param x
+     * @param meanLog
+     * @param sdLog
+     * @return 
+     */
+    template<class T>
+    const T dlnorm(const T &x, const T &meanLog, const T &sdLog, bool ret_log = false) {
+        if (x > T(0)) {
+            T ret = (T(1) / (x * sdLog * pow(T(2) * T(M_PI), 0.5))) *
```

Comment:
as above, better to write in logspace

Reply:
I think that there is a mistake in it as written with the denominator, shouldn't it be sigma^2?
```
ret = (T(1) / (x * sdLog * pow(T(2) * T(M_PI), 0.5))) *
                    exp(T(-1)*((std::log(x) - meanLog)*
                    (log(x) - meanLog)) / (T(2) * sdLog));
```
should be
```
ret = (T(1) / (x * sdLog * std::pow(T(2) * T(M_PI), 0.5))) *
                    exp(T(-1)*((std::log(x) - meanLog)*
                    (std::log(x) - meanLog)) / (T(2) * sdLog * sdLog));
```
and then taking the log of that led to the following but I think I made a mistake because I do not get the correct values in R if I code this equation 🤕 
```
ret = std::log(x * sdLog) - 0.5 * std::log(T(2) * T(M_PI)) - std::pow(std::log(x) - meanLog, 2) / (T(2) * sdLog * sdLog);
```

Reply:
Thanks @Andrea-Havron-NOAA, the negative is what was getting me 😮‍💨 . Also, @msupernaw let me know that the model will fail with `std::log(T(2))` and we need to use `fims_math::log(T(2))`. I will push your corrected version. Thanks again for the help. It was fun to remember the rules of logathrims 🤣 last night.

Reply:
@Andrea-Havron-NOAA I fixed this with b8d62505 but can you please check.